src/api/middleware.ts

```diff
@@ -174,96 +174,100 @@ export const LOG_STATUS_MESSAGE = "@{status}, user: @{user}(@{remoteIP}), req: '
 export const LOG_VERDACCIO_ERROR = `${LOG_STATUS_MESSAGE}, error: @{!error}`;
 export const LOG_VERDACCIO_BYTES = `${LOG_STATUS_MESSAGE}, bytes: @{bytes.in}/@{bytes.out}`;
 
-export function log(req: $RequestExtend, res: $ResponseExtend, next: $NextFunctionVer): void {
-  // logger
-  req.log = logger.child({ sub: 'in' });
+export function log(config: Config) {
+  return function (req: $RequestExtend, res: $ResponseExtend, next: $NextFunctionVer): void {
+    // logger
+    req.log = logger.child({ sub: 'in' });
+
+    const _auth = req.headers.authorization;
+    if (_.isNil(_auth) === false) {
+      req.headers.authorization = '<Classified>';
+    }
 
-  const _auth = req.headers.authorization;
-  if (_.isNil(_auth) === false) {
-    req.headers.authorization = '<Classified>';
-  }
+    const _cookie = req.headers.cookie;
+    if (_.isNil(_cookie) === false) {
+      req.headers.cookie = '<Classified>';
+    }
 
-  const _cookie = req.headers.cookie;
-  if (_.isNil(_cookie) === false) {
-    req.headers.cookie = '<Classified>';
-  }
+    req.url = req.originalUrl;
+    req.log.info({ req: req, ip: req.ip }, "@{ip} requested '@{req.method} @{req.url}'");
+    req.originalUrl = req.url;
 
-  req.url = req.originalUrl;
-  req.log.info({ req: req, ip: req.ip }, "@{ip} requested '@{req.method} @{req.url}'");
-  req.originalUrl = req.url;
+    if (_.isNil(_auth) === false) {
+      req.headers.authorization = _auth;
+    }
 
-  if (_.isNil(_auth) === false) {
-    req.headers.authorization = _auth;
-  }
+    if (_.isNil(_cookie) === false) {
+      req.headers.cookie = _cookie;
+    }
 
-  if (_.isNil(_cookie) === false) {
-    req.headers.cookie = _cookie;
-  }
+    let bytesin = 0;
+    if (_.get(config, 'experiments.bytesin_off') === true) {
```

Comment:
This logic is completely correct, it disables the whole bytes in unless is explicitly enabled.  Also better use optional chaining, I'm trying to get rid of `lodash`. 

```
    let bytesin = 0;
    if (_.isNil(config?.experiments?.bytesin_off) || config?.experiments?.bytesin_off === false) {
      req.on('data', function(chunk): void {
        bytesin += chunk.length;
      });
    }
```

Reply:
Yep, sure
Optional chaining is awesome

Reply:
Oh, sorry
My mistake, logic is absolutely incorrect

Reply:
that works for me even better.